crates/uopool/src/validate/mod.rs

```diff
@@ -51,47 +50,107 @@ pub enum UserOperationValidatorMode {
 /// The [UserOperation](UserOperation) validator trait.
 /// The [UserOperationValidator](UserOperationValidator) is a composable trait that allows bundler to choose validation rules(sanity, simultation, simulation trace) to apply.
 #[async_trait::async_trait]
-pub trait UserOperationValidator<P, R, E>: Send + Sync
-where
-    P: Mempool<UserOperations = VecUo, CodeHashes = VecCh, Error = E> + Send + Sync,
-    R: Reputation<ReputationEntries = Vec<ReputationEntry>, Error = E> + Send + Sync,
-{
-    async fn validate_user_operation(
+pub trait UserOperationValidator: Send + Sync {
+    async fn validate_user_operation<T, Y, X, Z, H, R>(
         &self,
         uo: &UserOperation,
-        mempool: &MempoolBox<VecUo, VecCh, P, E>,
-        reputation: &ReputationBox<Vec<ReputationEntry>, R, E>,
+        mempool: &Mempool<T, Y, X, Z>,
+        reputation: &Reputation<H, R>,
         mode: EnumSet<UserOperationValidatorMode>,
     ) -> Result<UserOperationValidationOutcome, ValidationError>
     where
-        P: Mempool<UserOperations = VecUo, CodeHashes = VecCh, Error = E> + Send + Sync;
+        T: UserOperationAct,
+        Y: UserOperationAddrAct,
+        X: UserOperationAddrAct,
+        Z: UserOperationCodeHashAct,
+        H: HashSetOp,
+        R: ReputationEntryOp;
 }
 
 /// The [UserOperation](UserOperation) sanity check helper trait.
-pub struct SanityHelper<'a, M: Middleware + 'static, P, R, E>
-where
-    P: Mempool<UserOperations = VecUo, CodeHashes = VecCh, Error = E> + Send + Sync,
-    R: Reputation<ReputationEntries = Vec<ReputationEntry>, Error = E> + Send + Sync,
-{
-    mempool: &'a MempoolBox<VecUo, VecCh, P, E>,
-    reputation: &'a ReputationBox<Vec<ReputationEntry>, R, E>,
-    entry_point: EntryPoint<M>,
+pub struct SanityHelper<'a, M: Middleware + 'static> {
```

Comment:
I think we don't need SanityHelper anymore since entry_point and chain can be acccessed through mempool

Reply:
Mempool doesn't provide this now after the refactoring.